lib/iris/plot.py

```diff
@@ -665,6 +686,7 @@ def _map_common(draw_method_name, arg_func, mode, cube, plot_defn,
     return plotfn(*new_args, **kwargs)
 
 
+@_locker
 def contour(cube, *args, **kwargs):
```

Comment:
I'm inclined to agree... it feels like this should be done when testing. We are mostly wrapping matplotlib calls, which I guess aren't thread safe either, so it seems consistent to lock our wrappers.